src/view/view.cpp

```diff
@@ -403,7 +403,13 @@ void wf::view_interface_t::set_fullscreen(bool full)
     if (!fullscreen && get_output() && needs_lowering)
         get_output()->workspace->add_view(self(), wf::LAYER_WORKSPACE);
 
-    this->emit_signal("fullscreen", nullptr);
+    view_fullscreen_signal data;
```

Comment:
I would like to document this signal in `signal-definitions.hpp` (the goal is to eventually have all signals documented, along with the appropriate types).

Also, I think we could leave out desired size, since it is not always the full geometry. Let's leave it "undefined" or "0,0 0x0"